OMCompiler/SimulationRuntime/c/simulation/solver/linearSystem.c

```diff
@@ -393,6 +389,18 @@ int freeLinearSystems(DATA *data, threadData_t *threadData)
       freeAnalyticJacobian(jacobian);
       /* Note: The Jacobian of data->simulationInfo itself will be free later. */
 
+    if (linsys[i].parDynamicData == NULL)
+    {
+      break;
+    }
+    else
+    {
+      for (j=0; j<omc_get_max_threads(); ++j)
+      {
+        free(linsys[i].parDynamicData[j].b);
```

Comment:
Good idea to set `b` to NULL. I normally use a function called `delete_null()`  (or `free_null()` in this case) to delete and set to null. Instead of plain `free()`

Reply:
You mean after it is freed?

Reply:
Ya. I can see a cople of free calls followed by null assign here. So I am guessing there is some possibility of double free in mind.

Reply:
When initializing struct `whateverTheNameWasForLinearSystems` I set all pointers in it to `NULL`. When it gets deleted I check if it was allocated at some point by checking if it `NULL`.
Setting it to NULL here as well can't hurt. I'm doing that for other variables as well.

Reply:
Okay. I am just pointing them out in case you have overlooked them.

Reply:
I think it shouldn't matter. `freeLinearSystems` should only be called once, at least at the moment.
I'm adding a few `NULL`s nonetheless they don't hurt and there are a few of them already.

Reply:
>  at least at the moment.
There is the catch. It is better to be safe. Save time for your future self ☜(ﾟヮﾟ☜)

Reply:
Yeah I can see myself cursing which idiot implemented this in a year only to do a git blame and find myself. :sweat_smile:

Reply:
> I think it shouldn't matter. freeLinearSystems should only be called once
Btw it is not just about double free. The worst is when you use it after delete. It might work sometimes and segfault sometimes. Depending on weather the data has been overwritten since free.

NULL after free gives you a consistent crash in those cases.